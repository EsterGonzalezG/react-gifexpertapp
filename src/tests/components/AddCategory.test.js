import { shallow } from 'enzyme';
import React from 'react';
import { AddCategory } from '../../components/AddCategory';
describe('AddCategory', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test('should be show component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should changes input', () => {
    const value = 'Hola mundo';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });
  });

  test('should information on submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clear input test', () => {
    const value = 'Hola mundo';
    const input = wrapper.find('input');
    input.simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    expect(input.prop('value')).toBe('');
  });
});
