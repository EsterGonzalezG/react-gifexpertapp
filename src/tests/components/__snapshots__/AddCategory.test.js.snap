// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should be show component 1`] = `
<form
  onSubmit={[Function]}
>
  <label>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
</form>
`;
